pkg/neotest/signer.go

```diff
@@ -44,6 +44,14 @@ type MultiSigner interface {
 	Single(n int) SingleSigner
 }
 
+// ContractSigner is an interface for contract signer.
+// `InvocationScript()` and `ScriptHash()` must be defined.
+type ContractSigner interface {
```

Comment:
Then could we create a constructor for it? Like other signers have, so that it won't be necessary to implement this interface on the user side?

Reply:
This can be derived from manifest if contract is on-chain, see 
https://github.com/nspcc-dev/neo-go/blob/fd7b7ffc134481bf639ed8daf451a4b6b6f990be/pkg/services/rpcsrv/server.go#L959-L963

Reply:
`emit.Any` is in https://github.com/nspcc-dev/neo-go/pull/3237.

Reply:
For different transactions we can create different signer, it should be simpler (= shorter and easier to reason about) than using a constructor for arguments. Nothing prohibits us from using 2 signers for the same contract in different transactions like this:
```
c1 := inv.WithSigners(NewContractSigner(h, arg1))
...
c2 := inv.WithSigners(NewContractSigner(h, arg2))
```

Reply:
Done.

Reply:
Currently [the first suggestion](https://github.com/nspcc-dev/neo-go/pull/3233/commits/2f960fce4e41493b818dc5bd48cfe5497edfb598#r1407786333) implemented, but https://github.com/nspcc-dev/neo-go/pull/3233/commits/2f960fce4e41493b818dc5bd48cfe5497edfb598#r1408808907 is also a good scheme, I'm not sure which one we need to keep. @roman-khimov, could you take a look?